docs/functions.adoc

```diff
@@ -0,0 +1,2750 @@
+== PartiQL Functions
+
+This document gives answer to following questions informed by PartiQL Kotlin's implementation of a resolved query plan.
+
+.Questions
+. What is a function; what is an operator?
+. How do we define functions and operators?
+. What is SPECIFIC?
+. What are type coercions?
+. What is type precedence?
+. How are functions resolved?
+. How are NULL and MISSING handled in functions?
+
+== Proposals
+
+=== Q1. What is a function; what is an operator?
```

Comment:
Does this definition apply to aggregate functions? seemingly [PostgreSQL](https://www.postgresql.org/docs/current/sql-createaggregate.html) and [SQL Server](https://learn.microsoft.com/en-us/sql/t-sql/functions/aggregate-functions-transact-sql?view=sql-server-ver16) maybe among others have separated those.